ANMN/NRSMAI_product/NRSMAI_product_from_matfile.py

```diff
@@ -0,0 +1,139 @@
+tem_mi = MaskedArray(data['tem_mi'], mask=np.isnan(data['tem_mi']))
+sal_mi = MaskedArray(data['sal_mi'], mask=np.isnan(data['sal_mi']))
+
+# convert time variables to days from our chosen epoch
+dt = input_epoch - output_epoch    # difference as a timedelta object
+dt = dt.total_seconds()/24./3600.  # difference in decimal days
+
+time_mar_t = data['time_mar_t'] + dt
+time_mar_s = data['time_mar_s'] + dt
+time_mi = data['time_mi'] + dt
+
+
+### Put original sampling variables onto a common time dimension
+
+# Create a single time variable
+time_set_t = set(time_mar_t)
+time_set_s = set(time_mar_s)
+time_set_all = set.union(time_set_t, time_set_s) # set of all unique timestamps
+time = list(time_set_all)
+time.sort()
+
+# Find indices to map temp & sal measurements onto this time dimension
+index_t = []
+index_s = []
+for tt in time_mar_t:
+    index_t.append(time.index(tt))
+for tt in time_mar_s:
+    index_s.append(time.index(tt))
+
+
+### Create netCDF file for original sampling times
+F = inc.IMOSnetCDFFile(attribFile=attrib_file)
+
+TIMESERIES =  F.createVariable('TIMESERIES', np.int32, ())
+TIMESERIES[:] = 1
+
+TIME = F.setDimension('TIME', time)
+TIME.units = output_epoch.strftime('days since %Y-%m-%d %H:%M:%S UTC')
+TIME.comment = "Where only the sampling date is available, a UTC time of 00:00 is assumed."
+
+DEPTH = F.setDimension('DEPTH', depth)
+
+LATITUDE = F.setVariable('LATITUDE', F.geospatial_lat_min, ())
+LONGITUDE = F.setVariable('LONGITUDE', F.geospatial_lon_min, ())
+
+TEMP = F.createVariable('TEMP', tem_mar.dtype, ('TIME','DEPTH'))
+TEMP[index_t,:] = tem_mar
+TEMP.coordinates = 'TIME DEPTH LATITUDE LONGITUDE'
+
+PSAL = F.createVariable('PSAL', sal_mar.dtype, ('TIME','DEPTH'))
+PSAL[index_s,:] = sal_mar
+PSAL.coordinates = 'TIME DEPTH LATITUDE LONGITUDE'
+
+F.geospatial_vertical_min = min(depth)
+F.geospatial_vertical_max = max(depth)
+F.updateAttributes()
+savedFile = F.standardFileName(datacode=data_code, product=product_name_original)
+print savedFile
+
+F.close()
+
+
+### Create netCDF for interpolated timeseries
+G = inc.IMOSnetCDFFile(attribFile=attrib_file)
+
+TIMESERIES =  G.createVariable('TIMESERIES', np.int32, ())
+TIMESERIES[:] = 1
+
+TIME = G.setDimension('TIME', time_mi)
+TIME.units = output_epoch.strftime('days since %Y-%m-%d %H:%M:%S UTC')
+DEPTH = G.setDimension('DEPTH', depth)
+
+LATITUDE = G.setVariable('LATITUDE', G.geospatial_lat_min, ())
+LONGITUDE = G.setVariable('LONGITUDE', G.geospatial_lon_min, ())
+
+TEMP = G.createVariable('TEMP', np.float32, ('TIME','DEPTH'))
+TEMP[:] = tem_mi
+TEMP.coordinates = 'TIME DEPTH LATITUDE LONGITUDE'
+
+PSAL = G.createVariable('PSAL', np.float32, ('TIME','DEPTH'))
+PSAL[:] = sal_mi
+PSAL.coordinates = 'TIME DEPTH LATITUDE LONGITUDE'
+
+G.geospatial_vertical_min = min(depth)
+G.geospatial_vertical_max = max(depth)
+G.updateAttributes()
+savedFile = G.standardFileName(datacode=data_code, product=product_name_interpolated)
+print savedFile
+
+G.close()
+
```

Comment:
extra line.

Only have comments on a cosmetic side
 * upper case variable instead of lower case, (https://github.com/aodn/data-services/issues/283)  for netcdf vars and F, G
 * double empty lines in a row

Otherwise it makes sence
I imagine most of the nc file is set with inc.IMOSnetCDFFile(attribFile=attrib_file)